libcontainerd/local/local_windows.go

```diff
@@ -312,6 +312,11 @@ func (c *client) createWindows(id string, spec *specs.Spec, runtimeOptions inter
 			return errors.New("device assignment is not supported for HyperV containers")
 		}
 		for _, d := range spec.Windows.Devices {
+			// Per https://github.com/microsoft/hcsshim/blob/v0.9.2/internal/uvm/virtual_device.go#L17-L18,
+			// these represent an Interface Class GUID.
+			if d.IDType != "class" && d.IDType != "vpci-class-guid" {
+				return errors.Errorf("device assignment of type '%s' is not supported", d.IDType)
```

Comment:
Although there are no current tests for this, I confirmed that it's hit with the existing tests before adding `vpci-class-guid` to be supported here. See da16170d50c5247ef618837cb29267fb731af8fb.

It'll take a bit more work to hit this code in a generic test, because every other IDType known to hcsshim is supported only in one of HyperV or Process isolation modes, ClassGUID is the only mechanism that works across both.

Reply:
As part of [exposing and documenting these constants from hcsshim](https://github.com/moby/moby/pull/43368#discussion_r835767737), the coverage may be increased to cover this case (contrasting with the c8d backend not failing), although that is also increasing the amount of hcsshim-driven behaviour that is being tested.